### PR TITLE
Rustup

### PR DESCRIPTION
**Pull request by RalfJung**

### CODE REVIEW

File: tests/pass/weak_memory/extra_cpp_unsafe.rs [hunk 1]
```diff
@@ -27,7 +26,7 @@ fn racing_mixed_atomicity_read() {
 
     let j2 = spawn(move || {
         let x_ptr = x as *const AtomicU32 as *const u32;
-        unsafe { std::intrinsics::atomic_load_relaxed(x_ptr) }
+        unsafe { x_ptr.read() }
```
Comment (RalfJung) [inline tests/pass/weak_memory/extra_cpp_unsafe.rs:29]: @cbeuw this looks like a wrong test to me? The test description says it is about "perfectly overlapping non-atomic and atomic reads", but both threads have been doing atomic loads... I now changed one to non-atomic.

Comment (cbeuw) [reply tests/pass/weak_memory/extra_cpp_unsafe.rs:29]: Hmmm yeah that was a mistake

**Pull request merged**